src/sagas/assignments.js

```diff
@@ -0,0 +1,71 @@
+import {all, call, put, select, takeEvery} from 'redux-saga/effects';
+
+import {
+  getCourses,
+  createClassroomAssignment,
+} from '../clients/googleClassroom';
+import {
+  createProjectSnapshot,
+} from '../clients/firebase';
+import {
+  assignmentCreated,
+  assignmentNotCreated,
+} from '../actions/assignments';
+import {
+  closeAssignmentCreator,
+  updateCourses,
+} from '../actions/ui';
+import {
+  getCurrentProject,
+} from '../selectors';
+import {generateTextPreview} from '../util/compileProject';
+import {createSnapshotUrl} from '../util/exportUrls';
+
+export function* openAssignmentCreator() {
+  const courses = yield call(getCourses);
+  yield put(updateCourses(courses));
+}
+
+export function* assignAssignment(
+  {payload: {selectedCourse, dateInput}}) {
+  yield* createAssignment(
+    selectedCourse,
+    dateInput,
+    'PUBLISHED',
+  );
+}
+
+export function* draftAssignment(
+  {payload: {selectedCourse, dateInput}}) {
+  yield* createAssignment(selectedCourse, dateInput, 'DRAFT');
+}
+
+function* createAssignment(selectedCourse, dateInput, assignmentState) {
+  const project = yield select(getCurrentProject);
+  const snapshotKey = yield call(createProjectSnapshot, project);
+  const assignmentData = {
+    workType: 'ASSIGNMENT',
+    courseId: selectedCourse,
+    dateInput,
+    url: yield call(createSnapshotUrl, snapshotKey),
+    title: yield call(generateTextPreview, project),
+    state: assignmentState,
+  };
+
+  try {
+    const assignment = yield call(createClassroomAssignment, assignmentData);
+    yield put(assignmentCreated(assignment));
+    yield put(closeAssignmentCreator());
```

Comment:
If successful assignment creation should cause the assignment modal to close, the UI reducer should consume `ASSIGNMENT_CREATED` and update its state accordingly, no?